unlock-app/src/components/content/LogContent.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import Head from 'next/head'
+import Layout from '../interface/Layout'
+import CreatorAccount from '../creator/CreatorAccount'
+import CreatorLog from '../creator/CreatorLog'
+import BrowserOnly from '../helpers/BrowserOnly'
+import GlobalErrorConsumer from '../interface/GlobalErrorConsumer'
+import { pageTitle } from '../../constants'
+import withConfig from '../../utils/withConfig'
+import * as UnlockTypes from '../../unlock'
+
+interface Props {
+  account: UnlockTypes.Account
+  network: UnlockTypes.Network
+  transactionFeed: UnlockTypes.Transaction[]
+  transactionMetadataMap: UnlockTypes.TransactionMetadataMap
+}
+
+export const LogContent = (props: Props) => {
+  const { account, network, transactionFeed, transactionMetadataMap } = props
+  return (
+    <GlobalErrorConsumer>
+      <Layout title="Creator Log">
+        <Head>
+          <title>{pageTitle('Log')}</title>
+        </Head>
+        {account && (
+          <BrowserOnly>
+            <CreatorAccount network={network} account={account} />
+            <CreatorLog
+              transactionFeed={transactionFeed}
+              transactionMetadata={transactionMetadataMap}
+            />
+          </BrowserOnly>
+        )}
+      </Layout>
+    </GlobalErrorConsumer>
+  )
+}
+
+export const readable = (type: UnlockTypes.TransactionType) => {
+  let parts = type.split('_')
+  return parts.map(part => part[0] + part.slice(1).toLowerCase()).join(' ')
+}
+
+export const mapStateToProps = (
+  { account, network, transactions }: { account: UnlockTypes.Account,
+                                        network: UnlockTypes.Network,
+                                        transactions: UnlockTypes.Transactions,
+  },
+  { config: { chainExplorerUrlBuilders } } : { config: { chainExplorerUrlBuilders: UnlockTypes.ChainExplorerURLBuilders } }
+) => {
+  const transactionFeed = Object.values(transactions).sort(
+    (a, b) => b.blockNumber - a.blockNumber
+  )
+
+  const transactionMetadataMap: UnlockTypes.TransactionMetadataMap = {}
+
+  transactionFeed.forEach(tx => {
+    transactionMetadataMap[tx.hash] = {
```

Comment:
now that I read this I wonder if instead of passing `transactionMetadataMap` we should not use a react component like `<TransactionMetadata>` which would show what we need to show? That would probably be cleaner?

Let's open an issue for this and do that in a separate PR (you can get rid of the strange `TransactionMetadataMap` type too!

Reply:
I would just as soon get rid of the concept of transaction metadata altogether. I think the right solution for the readable type name is to simply change the enum (as mentioned above). Assuming that is done, we're just trying to pass one additional piece of information, which I think we can do in a very lightweight manner.